Support for static member as the value of setting using `::` #73

MrMon3y opened the pull request:
@tsimbalar 

Support "Serilog.Sinks.SystemConsole.Themes.AnsiConsoleTheme::Grayscale" to access a static member for arguments of a complex type.

Solves issue #73

tsimbalar commented:
This looks pretty good 👍  

Just a few remarks : 
- I believe all the classes under the `*.TestDummies.*` are not used anywhere, so they could be removed
- to "prove" the correct behavior, it would be nice to update the *Sample* project so that it uses the *Console* sink (instead of *Literate Console*)  and a custom theme with the new `::` syntax.

MrMon3y commented:
@tsimbalar 

**PR Update**
- Removed unused `TestDummies`
- Updated *Sample* project to use the *Console* sink and added a custom theme with new `::` syntax.

Should this also be reflected in the `readme.md` ?

tsimbalar commented on sample/Sample/Sample.csproj:
```diff
@@ -16,6 +16,7 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="1.1.1" />
     <PackageReference Include="Serilog.Sinks.Async" Version="1.0.1" />
+    <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
     <PackageReference Include="Serilog.Sinks.Literate" Version="2.0.0" />
```
I think the package Serilog.Sinks.Literate is no longer required, because the sample app only use the Console Sink

tsimbalar commented on test/Serilog.Settings.Configuration.Tests/Support/ClassHierarchy.cs:
```diff
@@ -0,0 +1,22 @@
+﻿namespace Serilog.Settings.Configuration.Tests.Support
+{
+    public abstract class DummyAbstractClass
```
I think this class (and its subclasses) are no longer needed. If this is the case, the file can probably be removed :)

tsimbalar commented:
Looks good ! 
I've added a few more comments, but I think this is close to being done 👍 

Regarding the edit to the `README.md`, I think it will sooner or later be included as part of serilog/serilog#1053 so that won't be necessary for now.

Maybe a comment in the `README.md` of [Serilog.Sinks.Console](https://github.com/serilog/serilog-sinks-console) would be nice though, but I'll create a separate issue for that one.

MrMon3y commented:
@tsimbalar 

**PR Update**
- Removed stale PackageReference
- Removed stale Class

tsimbalar commented:
Thanks Douglas, this looks good to me ! 👍
@nblumhardt @skomis-mm  : shall I merge it ?

nblumhardt commented:
Looks good to me! I see we are really overdue for unification of the various configuration systems - quite a bit of duplication here, but 👍 for now.

skomis-mm commented:
Looks good! 👍
